src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -231,8 +233,32 @@ protected final void verifyWithInlineConfigParser(String filePath, String... exp
             throws Exception {
         final TestInputConfiguration testInputConfiguration =
                 InlineConfigParser.parse(filePath);
+        final DefaultConfiguration treeWalker =
+                new DefaultConfiguration(TreeWalker.class.getName());
         final DefaultConfiguration parsedConfig =
-                testInputConfiguration.createConfiguration();
+                testInputConfiguration.createConfiguration(treeWalker);
+        verifyViolations(parsedConfig, filePath, testInputConfiguration.getViolations());
+        verify(parsedConfig, filePath, expected);
+    }
+
+    /**
+     * Performs verification of the file with the given file path using specified configuration,
+     * and the array expected messages. Also performs verification of the config specified in
+     * input file. Also set the treeWalker configuration given as argument
+     *
+     * @param filePath file path to verify.
+     * @param expected an array of expected messages.
+     * @param treeWalker TreeWalker Configuration
+     * @throws Exception if exception occurs during verification process.
+     */
+    protected final void verifyWithInlineConfigParser(String filePath,
+                                                      DefaultConfiguration treeWalker,
```

Comment:
Non of other execution require this, but almost all of them are children of Treewalker
Please debug around https://github.com/checkstyle/checkstyle/blob/c47dc6e5b4f95db935a391ea527c4a0300b10ad3/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L119
To figure out why it works for others but for your Check

Reply:
I debug and found that currently there is no check that setting the property of the treeWalker.

Reply:
Ok I see it 
https://github.com/checkstyle/checkstyle/pull/12386/files#diff-74a37ae26b9b5489dc1aeb5346ee75381aa4291455c9b6e488f56ccd1c5d90d2L430

Reply:
But in this case we need something like https://github.com/checkstyle/checkstyle/pull/12386#issuecomment-1311950902

As all should be defined in Input file.
Please create one more PR and try to implement not adding traleewaler under Treewalker but to update properties of it from values from file.

Reply:
ok

Reply:
can you please create the Issue for this ?

Reply:
I am having some problem in handling the exception, what should I write in catch,

here is my function for processing each moduleconfig,

```
    public DefaultConfiguration createConfiguration() {
        final DefaultConfiguration root = new DefaultConfiguration(ROOT_MODULE_NAME);
        final DefaultConfiguration treeWalker =
                new DefaultConfiguration(TreeWalker.class.getName());
        root.addProperty("charset", StandardCharsets.UTF_8.name());
        childrenModules
                .stream()
                .map(ModuleInputConfiguration::createConfiguration)
                .forEach(moduleConfig -> {
                    processModuleConfigs(root, treeWalker, moduleConfig);
                });
        root.addChild(treeWalker);
        return root;
    }

    private static void processModuleConfigs(DefaultConfiguration root,
                                     DefaultConfiguration treeWalker,
                                     DefaultConfiguration moduleConfig) {
        if (CHECKER_CHILDREN.contains(moduleConfig.getName())) {
            root.addChild(moduleConfig);
        }
        else if (TREE_WALKER_MODULE_NAME.equals(moduleConfig.getName())) {
            for (String property : moduleConfig.getPropertyNames()) {
                try {
                    treeWalker.addProperty(property, moduleConfig.getProperty(property));
                } catch (CheckstyleException ex) {
                    
                }
            }
        }
        else {
            treeWalker.addChild(moduleConfig);
        }
    }
```

Reply:
Most likely this PR will win https://github.com/checkstyle/checkstyle/pull/12395